File: 2mingguakhir.r
library(dplyr)
library(lubridate)

now <- Sys.time()
print(now)

today <- today()
print(today)

hari_ini <- ymd('2024-03-21')
print(hari_ini)

waktu_ini <- hms('12:30:45')
print(waktu_ini)

two_weeks_ago1 <- now - weeks(2)
print(two_weeks_ago1)


two_weeks_ago2 <- today - weeks(2)
print(two_weeks_ago2)

two_weeks_ago3 <- hari_ini - weeks(2)
print(two_weeks_ago3)

df <- data.frame(
    Date = c("2024-02-25", "2024-03-16", "2024-02-01", "2024-05-10", "2024-04-01"),
    Terjual = c("Sapu", "Sapi", "Engkrak", "Nasgor", "sushi")
)
df$Date <- as.Date(df$Date)
print(df)
hasil <- df[df$Date >= two_weeks_ago2, ]
print(hasil)

File: 2minggu_akhir.py
from datetime import datetime, timedelta
import pandas as pd

now = datetime.now().date()
two_weeks_ago1 = now - timedelta(weeks=2)
two_weeks_ago2 = now - timedelta(days = 14)
print(two_weeks_ago1)
print(two_weeks_ago2)

two_weeks_ago = datetime.now() - timedelta(weeks=2)


df = pd.DataFrame({
    'Date': ['2024-02-25', '2024-03-16', '2024-02-01', '2024-05-10', '2024-04-01'],
    'Terjual': ['Sapu', 'Sapi', 'Engkrak', 'Nasgor', 'sushi']
})
df['Date'] = pd.to_datetime(df['Date'])
print(df)
df = df[df['Date'] >= two_weeks_ago]
print(df)